core/dns.go

```diff
@@ -70,7 +70,7 @@ func NewDNSResolverImpl(dialTimeout time.Duration, servers []string) *DNSResolve
 // This constructor should *only* be called from tests (unit or integration).
 func NewTestDNSResolverImpl(dialTimeout time.Duration, servers []string) *DNSResolverImpl {
 	resolver := NewDNSResolverImpl(dialTimeout, servers)
-	resolver.allowLoopbackAddresses = true
+	resolver.allowRestrictedAddresses = true
```

Comment:
This one's technically a rename since the functionality changed a bit. The 'allowLoopback' was introduced in https://github.com/letsencrypt/boulder/pull/739 because otherwise integration tests don't work (they rely on connecting to localhost).

This change expands the boolean so it allows all otherwise-forbidden networks, since Docker's host is not localhost but 172.*